### PR TITLE
[Docs][Apple] Update apple platform build docs

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Update tested OS+Xcode combinations to match the versions used in CI.
We dont have any other regular builds happen with other combinations, so Update
recommendations to match known working versions in line with CI.

## Motivation and context
Update docs to match CI to make it easy for new devs to be able to match the output of our CI system.

## What is the effect on users?
More consistent documentation with known working combinations according to our CI builds

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [X] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: docs/README.tvOS.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ![Kodi Logo](resources/banner_slim.png)
 
 # tvOS build guide
-This guide has been tested with macOS 10.13.4(17E199) High Sierra and 10.14.4(18E226) Mojave on Xcode 9.4.1(9F2000) and Xcode 10.2(10E125). It is meant to cross-compile Kodi for tvOS 11+ (AppleTV 4/4K) using **[Kodi's unified depends build system](../tools/depends/README.md)**. Please read it in full before you proceed to familiarize yourself with the build procedure.
+This guide has been tested with macOS 10.15.2 Mojave and Xcode 11.3.1. Please note this is the only version our CI system builds with. There is no assurances that other combinations will work correctly. It is meant to cross-compile Kodi for tvOS 11+ (AppleTV 4/4K) using **[Kodi's unified depends build system](../tools/depends/README.md)**. Please read it in full before you proceed to familiarize yourself with the build procedure.
```
Comment (garbear) [inline docs/README.tvOS.md:4]: intended to have tvOS on a different OS version than macOS and iOS builds?

Comment (fuzzard) [reply docs/README.tvOS.md:4]: the 3rd CI server that solely builds tvos uses Catalina. The other 2 are using Mojave as documented, so yes, at this time its intended.

Comment (kambala-decapitator) [reply docs/README.tvOS.md:4]: I suggest to keep Mojave (10.14.6) as I've successfully tested this setup multiple times. Btw 10.15 is Catalina :)

Comment (fuzzard) [reply docs/README.tvOS.md:4]: The intent is to keep it inline with whatever CI uses. I could say Catalina and Xcode 12, but what happens when I move on and that combination is no longer tested/used. At least with Jenkins it runs constantly and we can guarantee the combo works at any given point

Comment (fuzzard) [reply docs/README.tvOS.md:4]: Also, was a C&P error, I'll fix it

**Comment by kambala-decapitator** [comment]
I think we should clearly state the following:

1. minimum required OS to build using the Xcode version used on CI
2. recommended constellation that matches CI setup (already there)
3. any later versions would probably also work, but no guarantees that app would behave the same as the release version

**Comment by fuzzard** [comment]
As discussed, i hope all 3 are now answered in the docs with this wording.

**Review by kambala-decapitator — APPROVED** [review]

**Review by garbear — APPROVED** [review]

**Pull request merged**